specifyweb/backend/setup_tool/setup_tasks.py

```diff
@@ -0,0 +1,171 @@
+        return bool(res)
+    except Exception:
+        return False
+
+def get_active_setup_task() -> Tuple[Optional[AsyncResult], bool]:
+    """Return the current setup task if it is active, and also if it is busy."""
+    global _active_setup_task_id
+    with _active_setup_lock:
+        task_id = _active_setup_task_id
+
+    if not task_id:
+        return None, False
+
+    res = app.AsyncResult(task_id)
+    busy = res.state in ("PENDING", "RECEIVED", "STARTED", "RETRY", "PROGRESS")
+    # Check if the last task ended
+    if not busy and res.state in ("SUCCESS", "FAILURE", "REVOKED"):
+        # Get error message if any.
+        if res.state == "FAILURE":
+            info = getattr(res, "info", None)
+            if isinstance(info, dict):
+                error = info.get("error") or info.get("exc_message") or info.get("message") or repr(info)
+            else:
+                error = str(info)
+            set_last_setup_error(error)
+        # Clear the setup id if its not busy.
+        # Commented out to allow error messages to be checked multiple times.
+        # with _active_setup_lock:
+        #     if _active_setup_task_id == task_id:
+        #         _active_setup_task_id = None
+    return res, busy
+
+@app.task(bind=True)
+def setup_database_task(self, data: dict):
+    """Execute all database setup steps in order."""
+    self.update_state(state='STARTED', meta={'progress': api.get_setup_resource_progress()})
+    def update_progress():
+        self.update_state(state='STARTED', meta={'progress': api.get_setup_resource_progress()})
+
+    try:
+        with transaction.atomic():
+            logger.debug('## SETTING UP DATABASE WITH SETTINGS:##')
+            logger.debug(data)
+
+            logger.debug('Creating institution')
+            api.create_institution(data['institution'])
+            update_progress()
+
+            logger.debug('Creating storage tree')
+            api.create_storage_tree(data['storagetreedef'])
+            update_progress()
+
+            if data['institution'].get('issinglegeographytree', False) == True:
+                logger.debug('Creating singular geography tree')
+                api.create_global_geography_tree(data['globalgeographytreedef'])
+
+            logger.debug('Creating division')
+            api.create_division(data['division'])
+            update_progress()
+
+            discipline_type = data['discipline'].get('type', '')
+            is_paleo_geo = discipline_type in PALEO_DISCIPLINES or discipline_type in GEOLOGY_DISCIPLINES
+            default_tree = {
+                'fullnamedirection': 1,
+                'ranks': {
+                    '0': True
+                }
+            }
+
+            # if is_paleo_geo:
+            # Create an empty chronostrat tree no matter what because discipline needs it.
+            logger.debug('Creating Chronostratigraphy tree')
+            api.create_geologictimeperiod_tree(default_tree.copy())
```

Comment:
(optional / note)

I believe the default FullNameDirection for every tree _except_ GeologicTimePeriod is 1 (Forward).
GeologicTimePeriod has a default of -1 (Reverse). 

Do we want to handle that default in the `default_tree` when passing it to this function?